servo/connectors/kubernetes.py

```diff
@@ -1786,7 +1803,7 @@ async def rollout(self, *, timeout: Optional[servo.DurationDescriptor] = None) -
                     if event_type == "ERROR":
                         stream.stop()
                         # FIXME: Not sure what types we expect here
-                        raise servo.AdjustmentRejectedError(reason=str(deployment))
+                        raise servo.AdjustmentRejectedError(str(deployment))
```

Comment:
doesn't this need a reason code specified?

Reply:
We should model the behavior and reraise if its required

Reply:
Added start-failed as the reason to resolve this for the time being

@blakewatters can you provide more context on what you mean by modelling this out? I'm not sure of what context would even cause this code path to be hit so any additional details would be appreciated